Source/Testably.Abstractions.Testing/FileSystem/FileSystemWatcherMock.cs

```diff
@@ -353,17 +353,27 @@ private bool MatchesFilter(ChangeDescription changeDescription)
 			if (!changeDescription.Path.StartsWith(fullPath,
 				_fileSystem.Execute.StringComparisonMode))
 			{
-				return changeDescription.ChangeType == WatcherChangeTypes.Renamed &&
-				       changeDescription.OldPath?.StartsWith(fullPath,
-					       _fileSystem.Execute.StringComparisonMode) == true;
+				if (!(changeDescription.ChangeType == WatcherChangeTypes.Renamed &&
+				      changeDescription.OldPath?.StartsWith(fullPath,
+					      _fileSystem.Execute.StringComparisonMode) == true))
+				{
+					return false;
+				}
 			}
 		}
 		else if (!string.Equals(
 			_fileSystem.Execute.Path.GetDirectoryName(changeDescription.Path),
 			fullPath,
 			_fileSystem.Execute.StringComparisonMode))
 		{
-			return false;
+			if (!(changeDescription.ChangeType == WatcherChangeTypes.Renamed &&
+			      string.Equals(
```

Comment:
```suggestion
			if (changeDescription.ChangeType != WatcherChangeTypes.Renamed ||
			    !string.Equals(
```

Same here; I think the inteded logic reads more clearly.